partiql-ast/src/main/java/org/partiql/ast/WindowFunctionType.java

```diff
@@ -0,0 +1,160 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.expr.Expr;
+import org.partiql.ast.expr.ExprWindowFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a window function type.
+ * @see ExprWindowFunction#getFunctionType() 
+ */
+public abstract class WindowFunctionType extends AstNode {
+    /**
+     * A window function that takes no arguments.
+     * @see ExprWindowFunction#getFunctionType()
+     * @see WindowFunctionSimpleName
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class NoArg extends WindowFunctionType {
+        private final WindowFunctionSimpleName name;
+
+        /**
+         * Constructs a no-argument window function type.
+         * @param name the name of the window function
+         */
+        public NoArg(@NotNull WindowFunctionSimpleName name) {
+            this.name = name;
+        }
+
+        /**
+         * Returns the name of the window function.
+         * @return the name of the window function
+         */
+        @NotNull
+        public WindowFunctionSimpleName getName() {
+            return this.name;
+        }
+
+        @NotNull
+        @Override
+        public List<AstNode> getChildren() {
+            List<AstNode> kids = new ArrayList<>();
+            kids.add(name);
+            return kids;
+        }
+
+        @Override
+        public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
+            return visitor.visitWindowFunctionTypeNoArg(this, ctx);
+        }
+    }
+
+    /**
+     * Represents either the LAG or LEAD window function.
+     * @see ExprWindowFunction#getFunctionType()
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class LeadOrLag extends WindowFunctionType {
+        private final boolean isLead;
+        private final Expr extent;
+        private final Long offset;
+        private final Expr defaultValue;
```

Comment:
If window functions were modeled more closely to `ExprCall`, these could all be function arguments rather than hard-coded properties.